active/0030-modularize-app-boot.md

```diff
@@ -0,0 +1,40 @@
+- Start Date: 2015-12-21
+- RFC PR: [#30](https://github.com/ember-cli/rfcs/pull/30)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+Rework the `app-boot` hook to make it more module-friendly, reduce the API surface area, and make it simpler to unravel the Ember application programatically.
+
+# Motivation
+
+Treating `app-boot` as an appended section of imperative code inside of app.js makes it difficult to reason about in a modules-only world. We only have a few occurrences of this in our extended codebases (I audited every occurrence on GitHub), which implies that we can reduce the API surface area and accomplish the same underlying goal of triggering the initial `require` that unravels everything.
+
+* [ember-cli](https://github.com/ember-cli/ember-cli/blob/a2a9a886e146ce7bdfbdd5338542538ea7b38d14/lib/broccoli/ember-app.js#L1600) - Generates the initial require.
+* [ember-cli-delay-app-boot](https://github.com/rwjblue/ember-cli-delay-app-boot/blob/master/index.js#L31) - Throws a timeout around the initial require and includes some copypasta from above.
+* [ember-cli-fastboot](https://github.com/tildeio/ember-cli-fastboot/blob/2058dd1ebf08be818851d1e65651db49cdd0e69f/index.js#L73) - This adds a new specially-named module and doesn't call `require`. Instead, it itself becomes the module which must be required to boot the app.
+- In the future (read: I'm already encountering this) a Service Worker may wish to tell the application when it should boot or bind that functionality onto a promise chain. My current approach is to delay responding to the app-boot request and maintain that promise chain inside of the service worker.
+
+# Detailed design
```

Comment:
after reading the detailed designed, i am confused what is actually being proposed.